src/ModelContextProtocol.Core/Protocol/NotificationParams.cs

```diff
@@ -21,4 +21,21 @@ private protected NotificationParams()
     /// </remarks>
     [JsonPropertyName("_meta")]
     public JsonObject? Meta { get; set; }
+
+    /// <summary>
+    /// Merges additional metadata into the existing Meta object.
+    /// </summary>
+    /// <param name="additionalMeta">The additional metadata to merge.</param>
+    public void MergeMeta(JsonObject additionalMeta)
```

Comment:
We're needing to duplicate this method into each place that has Meta. We're also baking in the overwrite semantics of the additionalMeta always winning, as opposed to an existing value winning. If this is common enough to need a helper for, should we instead expose a static merge method?

@eiriktsarpalis, are we missing something here in STJ? What do you think we should do?